### ISSUE TITLE
Some styles not applied/missing in editor modern theme configuration

### ISSUE THREAD

**Issue by edmundito** [adventuregamestudio/ags#1974]
**Describe the bug**
The modern theme does not seem to be able to apply the following styles to the editor:

* script-editor/text-editor/indent-guide
* script-editor/text-editor/current-statement
* script-editor/text-editor/current-statement-2

Additionally, when a theme is applied, char values in the text editor appear to have a white background and black text, there seems to be no way to set the color values for a char.

**AGS Version**
3.6.0.47

**To Reproduce**
1. Install themes from https://github.com/adventuregamestudio/ags-themes
2. Select [C] VisualStudioDark and restart
3. Open the GlobalScript, note the indent guide
4. Select [M] Dark theme and restart
5. Open the GlobalScript, note the lack of an indent guide
6. Edit the [M] Dark theme and change the styles above to background: blue, foreground: red
7. Restart the editor

**Expected behavior**
The styles are visible

**Screenshots**
Char style missing, also showing lack of indent guide:
![image](https://user-images.githubusercontent.com/168664/230727056-795e1d1c-ef61-4c0a-a214-1869f7fe0c88.png)

**Comment by edmundito** [comment]
Another thing I noticed: When first loading the editor, some theme colors were not applied properly to the panel heading buttons (the ones in red below) Later on they do fix themselves (when I focus on each panel)

![image](https://user-images.githubusercontent.com/168664/230727939-cb3ba591-56ba-413f-87fd-0b022dcf0440.png)

**Comment by AlanDrake** [comment]
The "indent-guide" we're using are the vertical ones. They work but you only see them when indenting twice or more.
"current-statement" is the marker arrow. It works.
"current-statement-2" is a typo, it should be "current-statement2".

What doesn't work is overriding the foreground color of the marker line, but perhaps that should be done differently, because the MarkerSymbol Background is not doing anythign with the foreground color. Or it could be a scintilla bug, or we didn't implement that piece correctly. Needs investigation.

**Comment by ericoporto** [comment]
I think the colors not applying in pin and cross buttons is from here  https://github.com/dockpanelsuite/dockpanelsuite/issues/509

**Comment by ivan-mogilko** [comment]
Char style background was fixed by #2176, I will backport to 3.6.0 branch, as this is a trivial fix.

I did not look into the themes much, could anyone update me on this ticket, what would be required to fix these, and is this applicable to 3.6.0 release at all, or better left for later?

**Comment by AlanDrake** [comment]
This reminds me I need to do a round to add styles to new components, like the log panel, and update themes.
Panel which for some reason has slightly bigger icons than what it's supposed to have, so they appear blurry...

**Comment by ivan-mogilko** [comment]
Please, can anyone update on this ticket, is it done, if not then what remains?

**Comment by ericoporto** [comment]
https://github.com/adventuregamestudio/ags/issues/1974#issuecomment-1500907436

This comment is not feasible to fix without modifying the Docking component we use as it's not maintained anymore, or alternatively in the future maybe switching the component - but this isn't easy and the docking component that is better maintained doesn't try to emulate VS appearance.

**Comment by ivan-mogilko** [comment]
I propose to make a clear list of remaining problems here, and define whether they may or not be fixed within the 3.6.1,
then move those that require major changes to a separate ticket.

From what I understand, there were following issues mentioned here:
"Modern theme" styles not applied:
- script-editor/text-editor/indent-guide
- script-editor/text-editor/current-statement
- script-editor/text-editor/current-statement-2

Additional problem with styles:
- char values in the text editor appear to have a white background and black text - **This is fixed in 3.6.0 and 3.6.1**
- there seems to be no way to set the color values for a char [literal] - I separate this from above
- when first loading the editor, some theme colors were not applied properly to the panel heading buttons. Later on they do fix themselves (when I focus on each panel). - **too difficult to fix, postpone for later?**

**Comment by AlanDrake** [comment]
Indent styles are fine, "current-statement-2" was a typo, as I mentioned it's "current-statement2"

I'll take a look if I can style literals and add it to the themes.

The VS2005 theme not applying color for the heading buttons is a dockpanel issue, the maintainer said "Something happens to work does not mean it is supported." so it seems it was not intended to work unless you compiled your own theme with different colors.

**Comment by ericoporto** [comment]
> The VS2005 theme not applying color for the heading buttons is a dockpanel issue, the maintainer said "Something happens to work does not mean it is supported." so it seems it was not intended to work unless you compiled your own theme with different colors.

It's exactly that, apparently everyone just changes by code because it almost works fine, but they stated the following in that issue just above

> Changing colors via code is not supported. You have to change the color in the theme.

**Comment by AlanDrake** [comment]
Alright, 8ab2e0a implements themeable styles for "character", "brace-light", and "brace-bad".

**Comment by ivan-mogilko** [comment]
Alright, checking the mentioned issues.

1. `script-editor/text-editor/current-statement`, `script-editor/text-editor/current-statement-2`;
I had to go to the code to find out what that is. Appears that these are markers displayed when doing step-by-step execution (or stopping at breakpoints). One is an arrow and another a script line highlight. The correct name for the second is indeed `script-editor/text-editor/current-statement2`. I confirm that this works.

2. `script-editor/text-editor/indent-guide`;
These are thin lines appearing when you have blank space before your code. They are displayed **NOT** for **any** indent, but strictly once per each N spaces, following the "Tab width" setting in preferences. That is, if you set "tab width" to 2, they will be displayed each 2 spaces, if it's 8, they will be displayed each 8 spaces, regardless of how do you indent your code. On a side note, this preference setting is not reapplied dynamically, but only after you restart the Editor, but that's a separate issue.

3. *char values in the text editor appear to have a white background and black text*
Fixed in 3.6.0 and 3.6.1.

4. *way to set the color values for a char*; *"character", "brace-light", and "brace-bad".*
Implemented in 3.6.1, I confirm that this works.
I had to add these to a test theme by hand, existing themes do not feature these settings yet.

5. *panel heading buttons*
This is a dockpanel control issue. If it's a problem, then perhaps a separate ticket should be opened.
